vivisect/qt/funcgraph.py

```diff
@@ -277,6 +300,31 @@ def _hotkey_deczoom(self):
         #self.vw.vprint("NEW ZOOM    %f" % newzoom)
         self.mem_canvas.setZoomFactor(newzoom)
 
+    @pyqtSlot()
+    def refresh(self):
+        '''
+        Cause the Function Graph to redraw itself.
+        This is particularly helpful because comments and name changes don't
+        immediately display.  Perhaps someday this will update only the blocks
+        that have changed since last update, and be fast, so we can update
+        after every change.  
+        '''
+        self._last_viewpt = self.mem_canvas.page().mainFrame().scrollPosition()
+        self.clearText()
+        self.fva = None
+        self._renderDoneSignal.connect(self._refresh_cb)
+        self._renderMemory()
+
+    @workthread
+    @pyqtSlot()
+    def _refresh_cb(self):
+        '''
+        This is a hack to make sure that when _renderMemory() completes,
+        _refresh_3() gets run after all other rendering events yet to come.
+        '''
+        self.mem_canvas.setScrollPosition(self._last_viewpt.x(), self._last_viewpt.y())
+        self._renderDoneSignal.disconnect(self._refresh_cb)
+
```

Comment:
signals/slots get cleaned up by pyqt4 when objects go out of scope and get torn down. is there a particular behavior you're shooting for in disconnecting the signal/slot?  If not, they should just be connected during __init__ and allowed to tear down naturally :)